test/bats/e2e-provider.bats

```diff
@@ -0,0 +1,346 @@
+export SECRET_VALUE=${SECRET_VALUE:-"secret"}
+
+# export key vars
+export KEY_NAME=${KEY_NAME:-fookey}
+# defualt version value returned by mock provider
+export KEY_VERSION=${KEY_VERSION:-"v1"}
+# default key value returned by mock provider. 
+# base64 encoded content comparision is easier in case of very long multiline string.
+export KEY_VALUE_CONTAINS=${KEY_VALUE:-"LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KVGhpcyBpcyBtb2NrIGtleQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0K"}
+
+# export node selector var
+export NODE_SELECTOR_OS=$NODE_SELECTOR_OS
+# default label value of secret synched to k8s
+export LABEL_VALUE=${LABEL_VALUE:-"test"}
+
+@test "secretproviderclasses crd is established" {
+  kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  run kubectl get crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+  assert_success
+}
+
+@test "Test rbac roles and role bindings exist" {
+  run kubectl get clusterrole/secretproviderclasses-role
+  assert_success
+
+  run kubectl get clusterrole/secretproviderrotation-role
+  assert_success
+
+  run kubectl get clusterrole/secretprovidersyncing-role
+  assert_success
+
+  run kubectl get clusterrolebinding/secretproviderclasses-rolebinding
+  assert_success
+
+  run kubectl get clusterrolebinding/secretproviderrotation-rolebinding
+  assert_success
+
+  run kubectl get clusterrolebinding/secretprovidersyncing-rolebinding
+  assert_success
+}
+
+@test "deploy e2e-provider secretproviderclass crd" {
+  envsubst < $BATS_TESTS_DIR/e2e_provider_v1alpha1_secretproviderclass.yaml | kubectl apply -f -
+
+  kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/e2e-provider -o yaml | grep e2e-provider"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+}
+
+@test "CSI inline volume test with pod portability" {
+  envsubst < $BATS_TESTS_DIR/pod-secrets-store-inline-volume-crd.yaml | kubectl apply -f -
+  
+  kubectl wait --for=condition=Ready --timeout=180s pod/secrets-store-inline-crd
+
+  run kubectl get pod/secrets-store-inline-crd
+  assert_success
+}
+
+@test "CSI inline volume test with pod portability - read secret from pod" {
+  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl exec secrets-store-inline-crd -- cat /mnt/secrets-store/$SECRET_NAME | grep '${SECRET_VALUE}'"
+
+  result=$(kubectl exec secrets-store-inline-crd -- cat /mnt/secrets-store/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+}
+
+@test "CSI inline volume test with pod portability - read key from pod" {
+  result=$(kubectl exec secrets-store-inline-crd -- cat /mnt/secrets-store/$KEY_NAME)
+  result_base64_encoded=$(echo "${result//$'\r'}" | base64 ${BASE64_FLAGS})
+  [[ "${result_base64_encoded}" == *"${KEY_VALUE_CONTAINS}"* ]]
+}
+
+@test "CSI inline volume test with pod portability - unmount succeeds" {
+  skip
```

Comment:
Looks like we're skipping all the tests after `CSI inline volume test with pod portability - read key from pod` in the suite?

Reply:
done